src/common/JSHandle.ts

```diff
@@ -212,64 +182,57 @@ export class JSHandle<T = unknown> {
    * const children = [];
    * for (const property of properties.values()) {
    *   const element = property.asElement();
-   *   if (element)
+   *   if (element) {
    *     children.push(element);
+   *   }
    * }
    * children; // holds elementHandles to all children of document.body
    * ```
    */
   async getProperties(): Promise<Map<string, JSHandle>> {
-    assert(this.#remoteObject.objectId);
-    const response = await this.#client.send('Runtime.getProperties', {
-      objectId: this.#remoteObject.objectId,
-      ownProperties: true,
+    const keys = await this.evaluate(object => {
+      return Object.keys(object);
     });
-    const result = new Map<string, JSHandle>();
-    for (const property of response.result) {
-      if (!property.enumerable || !property.value) {
-        continue;
-      }
-      result.set(property.name, createJSHandle(this.#context, property.value));
-    }
-    return result;
+    const entries = await Promise.all(
+      keys.map(async key => {
+        const handle = await this.getProperty(key);
```

Comment:
Let's keep the existing logic here too. The previous version created all JSHandles with a single CDP call, this one does multiple leading to a potentially inconsistent data and performance impact,